pkg/cache/clusterqueue.go

```diff
@@ -627,3 +677,26 @@ func (c *ClusterQueue) UsedCohortQuota(fName kueue.ResourceFlavorReference, rNam
 
 	return cohortUsage
 }
+
+// DominantResourceShare returns a value from 0 to 100 representing the maximum of the ratios
+// of usage above nominal quota to the lendable resources in the cohort, among all the resources
+// provided by the ClusterQueue.
+// The function also returns the resource name that yielded this value.
+func (c *ClusterQueue) DominantResourceShare(w *workload.Info) (int, corev1.ResourceName) {
+	if c.Cohort == nil {
+		return 0, ""
+	}
+	var drs int64 = -1
+	var dRes corev1.ResourceName
+	wUsage := w.ResourceUsage()
+	for rName, rStats := range c.ResourceStats {
+		ratio := max(rStats.Usage+wUsage[rName]-rStats.Nominal, 0) * 100 /
+			c.Cohort.ResourceStats[rName].Lendable
```

Comment:
That would likely be a user error, but it could happen (you wouldn't try to share 0 resources in a cohort). Fixed and added unit test.

Reply:
I suppose there might be some legitimate scenarios. For example you may want to use sharing for one resource (say CPU), but set `lendingLimit=0` for GPU to prevent sharing of them, in a different resource flavor.